starter_kits/C/src/MyBot.c

```diff
@@ -0,0 +1,324 @@
+	// Now we start the game loop
+	while(1){
+		// Before we deal with the game engine, we should reset whether or not each map tile is occupied
+		for(i = 0; i < map_size[1]; i++){
+			for(j = 0; j < map_size[0]; j++){
+				map[j][i].occupied = (unsigned short)-1;
+			}
+		}
+
+
+		// Read the turn number
+		if(scanf("%hu", &turn_number) < 1){
+			exit(1);
+		}
+
+		// Now we're going to save the turn number to the log file just to show how it's done
+		fprintf(log_file, "Starting turn %hu\n", turn_number);
+		fflush(log_file);
+
+		// Read information about how each of the players are doing
+		for(i = 0; i < number_of_players; i++){
+			if(scanf("%hhu %hhu %hhu %u", &sca, &scb, &scc, &sia) < 4){
+				exit(1);
+			}
+
+			id = sca;
+			players[id].ship_count = scb;
+			players[id].dropoff_count = scc;
+			players[id].halite = sia;
+
+			// Parse ships
+			for(s = 0; s < players[id].ship_count; s++){
+				if(scanf("%hu %hhu %hhu %hu", &ssa, &sca, &scb, &ssb) < 4){
+					exit(1);
+				}
+
+				players[id].ships[s] = ssa;
+
+				ships[ssa].x = sca;
+				ships[ssa].y = scb;
+				ships[ssa].halite = ssb;
+
+				// And mark the map tile as occupied
+				map[sca][scb].occupied = ssa;
+			}
+
+			// And now parse drop offs
+			for(j = 0; j < players[id].dropoff_count; j++){
+				if(scanf("%hhu %hhu %hhu", &sca, &scb, &scc) < 3){
+					exit(1);
+				}
+
+				dropoffs[sca][0] = scb;
+				dropoffs[sca][1] = scc;
+				players[id].dropoffs[i] = sca;
+			}
+		}
+
+		// Now, the game engine gives us the number of map tiles that have an updated amount of halite
+		if(scanf("%hhu", &i) < 1){
+			exit(1);
+		}
+
+		// If there are any updates, we parse them and update the map
+		for(; i > 0; i--){
+			if(scanf("%hhu %hhu %hu", &sca, &scb, &ssa) < 3){
+				exit(1);
+			}
+
+			map[sca][scb].halite = ssa;
+		}
+
+		// Now we add logic for controlling the ships here
+		// Here, we call the default move function from default_logic.c for each of our ships
+		for(i = 0; i < players[my_id].ship_count; i++){
+			// Let's borrow ssa for the ship's ID just to make the code a little easier
+			ssa = players[my_id].ships[i];
+			naive_navigate(ssa);
+
+			// Let's also show how far each ship is away from the ship yard
+			fprintf(log_file, "Ship %hu is %hhu moves away from the ship yard!\n", ssa, calc_dist(ships[ssa].x, ships[ssa].y, players[my_id].ship_yard[0], players[my_id].ship_yard[1]));
+			fflush(log_file);
+		}
+
+		// Then we add logic for when to spawn ships
+		// The default "smart" logic will be to spawn a ship every 5th turn until we run out of halite or pass turn 50
+		if(turn_number % 5 == 0 && turn_number <= 50 && players[my_id].halite >= ship_cost){
+			spawn_ship();
+		}
+
+
+		// Finally, we print a new line, and then flush the stdout buffer so the game engine receives our commands
+		printf("\n");
+		fflush(stdout);
+
+		// And write another log stating it's the end of the turn!
+		fprintf(log_file, "Ending turn %hu\n\n", turn_number);
+		fflush(log_file);
```

Comment:
You are flushing many times and this is an expensive operation.  Can you just flush at the bottom?